proto/content_analysis/sdk/analysis.proto

```diff
@@ -129,3 +129,14 @@ message ContentAnalysisResponse {
 
   reserved 2 to 3;
 }
+
+// Generic message sent from Chrome to Agent.
+message AgentRequest {
+  optional ContentAnalysisRequest request = 1;
+  optional ContentAnalysisAcknowledgement ack = 2;
+}
```

Comment:
Two questions:
1. Are ContentAnalysisRequest and ContentAnalysisAck mutually exclusive? 
2. ContentAnalysisAck doesn't seem to fall under the request category?

Reply:
The sequence of steps is like this:
1/ Agent is told about a new request by giving them a ContentAnalysisRequest.  This request has a `request_token` field that uniquely identifies the request.
2/ Agent sends a response to chrome with a verdict.  This response uses the same `request_token` so that can map the response to the initiating request.
3/ Chrome handles the response and the sends an ack to the agent to let it know what it did.  For example, chrome honoured the verdict as specified, or the verdict was ignored because it was received after the timeout, etc.  Chrome uses the same `request_token` in the ack so that the agent knows which request/response is being ack'ed.

I'm not sure I understand "fall under the request category".

Reply:
Got it.  I was taking an agent centric view, so one is a "request sent to the agent" and the other a "response sent from the agent".  To keep things shorter, and you good with "ChromeToAgent" and "AgentToChrome" ?